src/dll/edbgEcmdDll.C

```diff
@@ -661,7 +661,11 @@ uint32_t dllInitDll() {
     gEDBG_HOME = realPath;
   }
 
+#ifdef EDBG_ISTEP_CTRL_FUNCTIONS
```

Comment:
Please add a comment block in the code to explain why we need to have a completely different init path when istep control is enabled?
Also, is that actually when you intend?  Do you want someone doing a getscom to call ipl_init just because istep code is compiled into their executable?

Reply:
Ok. Yes, its intended change for now. otherwise, there will be double initialization and pdbg asserts like explained in the commit message. I will add a note. please see the call flow of ipl_init() it already does pdbg_target_init

https://github.ibm.com/phal/ipl/blob/master/libipl/libipl.C#L108 -> https://github.ibm.com/phal/ipl/blob/master/libipl/libipl.C#L121 -> https://github.ibm.com/phal/libekb_p10/blob/master/libekb.C#L70

Reply:
Couple of points:

This change isn't "for now" - Based on the design ipl_init() will always call pdbg_targets_init(). At the moment ipl_init() doesn't do much else so doing it for just a getscom should be ok.

> otherwise, there will be double initialization and pdbg asserts like explained in the commit message

I can't see that in the commit message. As you mentioned it looks like you've merged this commit with an unrelated one though, so perhaps that's why the commit message disappeared. You will need to resplit this one out, let me know if you need some help doing so.

Reply:
I also also have issue with this - we shouldn't have to be calling a function labeled `ipl_init` to be able to perform basic hardware operations (i.e. `getscom`).  It makes the code unclear and makes determining ownership of the edbg init path difficult.  It seems like this need to be refined.

Why can't edbg continue to init pdbg since it is the loader of pdbg?  Then, if further on in the edbg execution we know we are calling istep functions, those then do the remaining ipl_init?

What you want to change here is the very start of the edbg load and that needs to be as clear as possible to not have unintended side effects.  At a minimum there needs to be a much better commit message explaining all the considerations and angles that went into wanting such a change.

Reply:
with the earlier approach, the istep execution is crashing because pdbg_target_init gets called twice (once in dllInit and second-time ipl_init() called as part of istep functions and because of that there is double initialization ). 

I tested this code . There is no issues with this approach

Reply:
Just because it works doesn't mean it is the right answer.  I still need answers to my questions and my comments addressed before I'll approve.

Reply:
Agreed the Jason. Looks like not a clean approch. considering use case of IPL time hw access use case. This code is going to fail any hw access during ipl time. like 
ipl_init(IPL_HOSTBOOT) updates istep_mode attribute to istep mode.  The design direction to restrict attribute updates in hostboot ipl window.  Currently not enabled in the code.